File: c++/problem-001/problem1.cpp
#include <iostream>
using std::cout;
using std::endl;

int main(){
    int sum = 0, size = 1000;
    for(int i=0; i < size; i++){
        if(i % 3 == 0 or i % 5 == 0){
            sum += i;
        }
    }

    cout << "sum of multiples of 3 or 5 below " << size << " is: " << sum << endl;
}

File: problem-001/js/problem1.js
var isMultiple = function (x, y) {
    return x % y === 0;
};

var findMultiples = function(x, y, until) {
    var i = 1;
    var multiples = [];

    for(i; i < until; i++) {
        if (i % x === 0 || i % y === 0) {
            multiples.push(i);
        }
    }

    return multiples;
};

var sum = function(numbers) {
    var _sum = 0;
    var n;

    for each (n in numbers) {
        _sum += parseInt(n, 10);
    }

    return _sum;
};

var sumMultiples = function(x, y, until) {
    var multiples = findMultiples(x, y, until);
    return sum(multiples);
}
